backend/dataall/modules/dataset_sharing/services/dataset_alarm_service.py

```diff
@@ -19,9 +19,7 @@ def trigger_table_sharing_failure_alarm(
             target_environment: Environment,
     ):
         log.info('Triggering share failure alarm...')
-        subject = (
-            f'ALARM: DATAALL Table {table.GlueTableName} Sharing Failure Notification'
-        )
+        subject = f'ALARM: DATAALL Table {table.GlueTableName} Sharing Failure Notification'[:100]
```

Comment:
If we switch to static subjects then I would remove the 100 character truncate and I would update this code in share processors:

```
except Exception as e:
                self.handle_revoke_failure(share_item=share_item, table=table, error=e)
                new_state = revoked_item_SM.run_transition(ShareItemActions.Failure.value)
                revoked_item_SM.update_state_single_item(self.session, share_item, new_state)
                success = False
```

At the very minimum the code inside self.handle_revoke_failure should be in try catch.. Email sending even if the entire service is down should never leave shares in progress. We can just log that sending email failed... Thats far better than leaving a share blocked.

I would argue that even table state updates should be in their own try/catch with retries if the aurora service is down or behaving erratically. 

How I would resolve this:

1. put the code inside  self.handle_revoke_failure into a try/catch so that this method can never affect state transitions and just log if it fails .. it's not important enough to break transitions.

2. revoked_item_SM.run_transition does not throw exceptions so we're fine with it. no changes needed.

3. revoked_item_SM.update_state_single_item this could be wrapped inside in a try/catch or even outside .. but after catching it should LOG and re-raise the exception.

@anushka-singh One thing I don't understand .. If an exception happens in this EXCEPT block does it get logged or not?

Reply:
Good catch Zilvinas! One solution is the proposed: adding a try/catch inside the `handle_revoke_failure` (and approve) in `dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py` to ensure that no matter what, when there is a failure the item ends up in `Share_Failed` status. 


Another option is to move the code a bit and place the notification after the update of the item (which makes more sense logically)

```
            except Exception as e:
                new_state = revoked_item_SM.run_transition(ShareItemActions.Failure.value)
                revoked_item_SM.update_state_single_item(self.session, share_item, new_state)
                success = False
                self.handle_revoke_failure(share_item, table, e)

```
Let me know your thoughts. We can also implement both

Reply:
I have implemented both solutions in next revision. Please check @dlpzx

Reply:
@zsaltys 
if there's an issue within self.handle_revoke_failure that raises another exception, it won't be caught by the except Exception as e block in process_revoked_shares. Instead, it will propagate up the call stack, and if not caught elsewhere, it could lead to the program terminating abruptly. To handle exceptions within self.handle_revoke_failure, we could include a try-except block within that method itself. Something like this:
```
def handle_revoke_failure(
        self,
        table: DatasetTable,
        share_item: ShareObjectItem,
        error: Exception,
) -> bool:
    try:
        logger.error(
            f'Failed to revoke S3 permissions to table {table.GlueTableName} '
            f'from source account {self.source_accountid}//{self.source_region} '
            f'with target account {self.target_environment.AwsAccountId}/{self.target_environment.region} '
            f'due to: {error}'
        )
        DatasetAlarmService().trigger_revoke_table_sharing_failure_alarm(
            table, self.share, self.target_environment
        )
        return True
    except Exception as inner_exception:
        # Log the inner exception, handle it, or raise it again if needed
        logger.error(f'Error in handle_revoke_failure: {inner_exception}')
        # Handle or raise the exception as needed

```